src/Maker/MakeAuthenticator.php

```diff
@@ -249,19 +264,35 @@ private function generateAuthenticatorClass(array $securityData, string $authent
             'Entity\\'
         );
 
-        $this->generator->generateClass(
-            $authenticatorClass,
-            'authenticator/LoginFormAuthenticator.tpl.php',
-            [
+        $userRepoFullyQualifiedName = null;
+        $userRepoName = null;
+
+        $template = 'authenticator/LoginFormAuthenticator.tpl.php';
+
+        if ($this->useSecurity51) {
+            $template = 'authenticator/Security51LoginFormAuthenticator.tpl.php';
+            $userRepoClassNameDetails = $this->generator->createClassNameDetails(
+                $userClassNameDetails->getShortName().'Repository',
+                'Repository'
+            );
+
+            if (class_exists($userRepoClassNameDetails->getFullName())) {
+                $userRepoFullyQualifiedName = trim($userRepoClassNameDetails->getFullName(), '\\');
+                $userRepoName = $userRepoClassNameDetails->getShortName();
+            }
+        }
+
+        $this->generator->generateClass($authenticatorClass, $template, [
                 'user_fully_qualified_class_name' => trim($userClassNameDetails->getFullName(), '\\'),
+                'user_repository_fully_qualified_class_name' => $userRepoFullyQualifiedName,
```

Comment:
The idea is to use the `UserRepository::class` created by `make:user` directly instead of leaning on an `EntityManagerInterface` instance to get the user as we currently do with the "old" login form authenticator. This of course introduces its own set of challenges if a `UserRepository` (or whatever the name of the repository is) does not exist. Thoughts on moving in this direction?

Granted, I have not taken into account the new features in 5.2 that @wouterj mentioned w/ the lazy user loading at the time of this comment.

Reply:
Yea, I would prefer to inject the repository class if there is a custom repository class. But, if we do that, we will need to:

A) Read the Doctrine metadata to see if there *is* a custom repository (and get its class name if there is one).
B) If there is *not* one, fallback to using EntityManagerInterface. 

Needing to do (B) is kind of annoying... but it also seems aggressive to throw an error if the user doesn't have a custom `UserRepository`. So, probably we should do it / try it.

Reply:
I'm not very sure about the policy of MakerBundle, but wouldn't it be possible to only support the experimental authenticator maker in Symfony 5.2? That would make this logic a lot easier, as this logic is done by the Security system. All that's needed is to pass the username to the password and configure the user provider (as is done in the "legacy" part of the maker).

Reply:
@wouterj ya you're correct, thats what im planning on doing for 5.2. But where it gets tricky is in 5.1 - i started working on implementing this way back when and time came hard to come by. So now we're in a situation where `<= 5.0` we have the original `make:auth`, `=== 5.1` we'll have this implementation as it written now, and in `>= 5.2` we'll use those new features. Hopefully `>5.3` there wont be any significant changes.. But in the meantime I think this is where we're at with this.

when we merge this the goal is to have `make:auth` work in all 3 scenerios

Reply:
Yeah, I understand (especially the lack of time bit).

What I'm wondering is whether the `5.1` scenario should be focused upon. Or if we can skip it and support the original maker in `<=5.1` and the new experimental authenticator in `5.2`. Given that 5.2 is going to released in just 2 weeks from now, and 5.1 reaching end of life in just 2.5 months from now, is anyone going to use the maker in a Symfony 5.1 project? (open question, this is my first MakerBundle PR review, so I honestly don't know)

Reply:
I'm with ya - I am chewing on the idea myself.. I would almost guarantee if we did skip support in 5.1, someone would open an issue in the coming months about it. That aside - 

Pro's of skipping 5.1 support:
- easier to maintain in the future
- quicker to get out the door now

Cons:
 - Loosing out on the ability to coin `Security51` (Pop culture reference to Area 51 here in the states).
 - Potentially cutting off / alienating a segment of "less experienced" users who are stuck on 5.1 for whatever reason in the future.
 - walking back the functionality that I've written to support 5.1 then saying we didn't want to support this functionality because it makes life easier.

That last 2 cons are the biggest things holding me back from coming out and saying yup, lets skip it. But that is not to say I'm not completely against the idea considering it would make life easier going forward. 

Curious as to what others think as well.

Reply:
Supporting only 5.2 would be very cool for me